### PR TITLE
Add support for file based module in bundle

### PR DESCRIPTION
**Pull request by b4nst**
Hey folks! This PR brings local module support to Bundles. Since it's quite a refactor, I'd really appreciate your input before I go any further. Hence why it's lacking tests, documentation and such.

> [!IMPORTANT]
> Heads up, this PR involves some hefty changes to the module fetcher.

While putting together the changes we discussed privately and in #268, I realized it's going to bulk up the current `Fetcher` module quite a bit. So, I've gone ahead and created a new `fetcher` package within the `engine`. This should make things more manageable and set the stage for adding new fetcher types down the road (like an HTTPS fetcher). What do you think, @stefanprodan?

At the moment, the `Local` fetcher only handles directories (think `timoni apply`). I didn't want to cram too much into one PR, so I've held off on adding `.tar.gz` file support for now. I'll tackle that in a separate PR.

I've given it a test spin with a modified version of podinfo, which I've linked here for you to try out (it uses the local redis). If this approach aligns with our goals, it could be a winner.

Before I mark this PR as ready, I'll beef up the testing (especially in the new fetcher package), flesh out the documentation, and tidy up the `BundleBuilder` for easier maintenance. I'd love to hear your thoughts on these next steps!

### CODE REVIEW

File: internal/engine/fetcher/local.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+func (f *Local) Fetch() (*apiv1.ModuleReference, error) {
+	if fs, err := os.Stat(f.src); err != nil || !fs.IsDir() {
+		return nil, fmt.Errorf("module not found at path %s", f.src)
+	}
+
+	modFile := path.Join(f.src, "cue.mod", "module.cue")
+	timoniFile := path.Join(f.src, "timoni.cue")
+	valuesFile := path.Join(f.src, "values.cue")
```
Comment (stefanprodan) [inline internal/engine/fetcher/local.go:60]: Aren't these declared in ` NewLocal` as requiredFiles?

File: examples/bundles/podinfo.cue [hunk 2]
```diff
@@ -8,11 +8,7 @@ bundle: {
 	name:       "podinfo"
 	instances: {
 		redis: {
-			module: {
-				url:     "oci://ghcr.io/stefanprodan/modules/redis"
-				version: "7.2.2"
-				digest:  "sha256:22ab5f4f0158f02b353c168f21a54901a45b2a56865fcc470e5492120deb4111"
-			}
+			module: url: "file://../redis"
```
Comment (stefanprodan) [inline examples/bundles/podinfo.cue:11]: I would undo this change and properly document this feature in https://timoni.sh/bundle/#working-with-bundles under a section called "Using local modules".

**Review by stefanprodan — COMMENTED** [review]
This looks great @b4nst just a couple of comments from me

File: internal/engine/fetcher/local.go (hunk 1 above)
Comment (b4nst) [reply internal/engine/fetcher/local.go:60]: Yes indeed ahaha. I declared them exactly for that purpose 🤦 . I'll cleanup that.

File: examples/bundles/podinfo.cue (hunk 2 above)
Comment (b4nst) [reply examples/bundles/podinfo.cue:11]: Sounds fair! Wilco

File: internal/testutils/registry.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	if err != nil {
+		t.Fatalf("failed to create docker registry: %s", err)
+	}
+
+	go dockerRegistry.ListenAndServe()
+
+	return address
+}
+
+// func SetupTestRegistry(t *testing.T) string {
```
[the remark this answers is not in the document]
Comment (b4nst) [reply internal/testutils/registry.go]: Oops thanks for catching that, done

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @b4nst this is an awesome feature 🏅 

PS. Going to add e2e tests in CI for local bundles in a followup PR.
